### PR TITLE
ENH(RF): initial RFing to bring control over colors/formatting, add highlighting to search results

### PR DESCRIPTION
**Pull request by yarikoptic**
Example:
screenshot
and should have none color/bold/underlying decoration  in non-interactive mode

### CODE REVIEW

File: datalad/metadata/search.py [hunk 1]
```diff
@@ -320,7 +325,8 @@ def pretty_str(s):
                            for k, v in s.items()])
     elif isinstance(s, text_type):
         try:
-            return s.encode('utf-8')
+            b = s.encode('utf-8')
+            return (os.linesep + "  ").join(_lines_regex.split(b))
```
Comment (yarikoptic) [inline datalad/metadata/search.py]: note to myself... we really need tests for cmdline rendering calls... I am even thinking if somehow we should automate at least smoke testing of them for any call of the Python API interface

Comment (mih) [inline datalad/metadata/search.py]: I am in favor of that, but didn't have the brilliant idea yet to do all in one go.

**Comment by mih** [comment]
Still WiP?

**Comment by yarikoptic** [comment]
if noone minds -- I would prefer to merge it for now as is... may be some time later would do a bigger overhaul and improvements (e.g. tests)

**Pull request merged**